### PR TITLE
configdrive: correct network config reading and improve tests

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

**Comment by marineam** [comment]
LGTM bugfix wise, haven't reviewed the test suite yet.

File: datasource/test/filesystem.go [hunk 1]
```diff
@@ -15,14 +15,40 @@
...
+
+func NewMockFilesystem(files ...File) MockFilesystem {
+	fs := MockFilesystem{}
+	for _, file := range files {
+		fs[file.Path] = file
+
+		// Create the directories leading up to the file
+		p := path.Dir(file.Path)
+		for p != "/" && p != "." {
+			fs[p] = File{Path: p, Directory: true}
```
Comment (jonboulle) [inline datasource/test/filesystem.go]: Maybe for sanity you should panic if `f, ok := fs[p]; ok && !f.Directory`

**Comment by jonboulle** [comment]
squid approved

**Comment by crawford** [comment]
@jonboulle good idea. done.

**Pull request merged**